### ISSUE TITLE
Custom gpu drivers

### ISSUE THREAD

**Issue by ROMANIOSerror** [Trixarian/NetherSX2-patch#91]
Is there a possibility to put in a custom gpu drivers like mesa turnip for vulkan or panVK for mali as a lot of crashes I experience was due to gpu driver issues 

I saw some emulators implement turnip custom drivers and it works just perfect so maybe Nethersx2 can benefit from it

**Comment by ROMANIOSerror** [comment]
![Screenshot_20240615-190025](https://github.com/Trixarian/NetherSX2-patch/assets/102884791/7945cdb6-2d28-408b-a408-9ddbf61c26ca)

**Comment by Trixarian** [comment]
PanVK probably won't work out quite that well. They already dropped Mali Midgard support, with Bifrost also on chopping block since they have a preference for Vahall and up (since it's less work getting Vulkan compliant)

**Comment by ROMANIOSerror** [comment]
But they are still working on it and may be they will keep Bifrost on the bus. Midgard was already old enough even when I saw it I wondered how they would put a custom driver on a 10 year old architectures

Yet I was wondering if you are able to implement custom gpu drivers like turnip without the source code

**Comment by ROMANIOSerror** [comment]
![Screenshot_20240623-045644](https://github.com/Trixarian/NetherSX2-patch/assets/102884791/78923076-cd83-44f4-b516-d610259a0a69)
And according to this photo I am not sure what does that mean exactly

Does it mean that no one tried to push custom drivers to mali or just it was not worth it as I was wondering if we got a similarity here because as far as I have read right now aethersx2 used custom adreno drivers but not mali drivers and I don't fully understand why 

So I was wondering can someone put custom mali drivers based on this information or is it hard and unworthy anyway and consider enlightening me about that topic because it is literally a little dark out there xD I guess as a medical student I am a little dumb for these topics xD

**Comment by Trixarian** [comment]
Yeah, the Vulkan errors is caused by the PCSX2 core using newer Vulkan calls than what the oem Vulkan drivers supports. It's probably worse by now with how deeply the PCSX2 guys have leaned into Vulkan's feature set. We'll need the higher version support for the more graphically intense games if we wish to use the newer cores
